laser_encoders/download_models.py

```diff
@@ -0,0 +1,155 @@
+#!/bin/bash
+# Copyright (c) Facebook, Inc. and its affiliates.
+# All rights reserved.
+#
+# This source code is licensed under the BSD-style license found in the
+# LICENSE file in the root directory of this source tree.
+#
+# LASER  Language-Agnostic SEntence Representations
+# is a toolkit to calculate multilingual sentence embeddings
+# and to use them for document classification, bitext filtering
+# and mining
+#
+# -------------------------------------------------------
+#
+# This python script installs NLLB LASER2 and LASER3 sentence encoders from Amazon s3
+
+# default to download to current directory
+
+import argparse
+import logging
+import os
+import sys
+from pathlib import Path
+
+import requests
+from language_list import LASER2_LANGUAGE, LASER3_LANGUAGE, SPM_LANGUAGE
+from tqdm import tqdm
+
+assert os.environ.get("LASER"), "Please set the environment variable LASER"
+LASER = os.environ["LASER"]
+sys.path.append(LASER)
+
+from laser_encoders.laser_tokenizer import LaserTokenizer
+from laser_encoders.models import SentenceEncoder
+
+logging.basicConfig(
+    stream=sys.stdout,
+    level=logging.INFO,
+    format="%(asctime)s | %(levelname)s | %(name)s | %(message)s",
+)
+logger = logging.getLogger("preprocess")
+
+
+class LaserModelDownloader:
+    def __init__(self, model_dir: str):
+        self.model_dir = model_dir
+        self.url = "https://dl.fbaipublicfiles.com/nllb/laser"
+
+    def download(self, url: str):
+        if os.path.exists(os.path.join(self.model_dir, os.path.basename(url))):
+            logger.info(f" - {os.path.basename(url)} already downloaded")
+        else:
+            logger.info(f" - Downloading {os.path.basename(url)}")
+            response = requests.get(url, stream=True)
+            total_size = int(response.headers.get("Content-Length", 0))
+            progress_bar = tqdm(total=total_size, unit="KB")
+            with open(os.path.join(self.model_dir, os.path.basename(url)), "wb") as f:
+                for chunk in response.iter_content(chunk_size=1024):
+                    f.write(chunk)
+                    progress_bar.update(len(chunk))
+            progress_bar.close()
+
+    def download_laser2(self):
+        self.download(f"{self.url}/laser2.pt")
```

Comment:
Instead of needing to pass `self.url` each time to the download script, perhaps refactor the `download` function to instead take as an argument the filename, and then use `self.url` to append to this for downloading.